qemu/tests/slof_multi_devices.py

```diff
@@ -0,0 +1,94 @@
+"""
+slof_multi_devices.py include following case:
+    1. SLOF boot successfully when adding two pci-bridge to the guest
+    2. VM boot successfully with lots of virtio-net-pci devices
+"""
+import logging
+
+from virttest import error_context
+from provider import slof
+from virttest import utils_net
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify SLOF info with multi-devices.
+    Step:
+        1. Boot a guest bu following ways:
+            a. attach two pci-bridges
+            b. multiple virtio-net-pci devices attached multiple pci-bridges
+               by one to one.
+        2. Check if any error info from output of SLOF.
+        3. Guest could login sucessfully.
+        4. Guest could ping external host ip.
+        5. For virtio-net-pci scenario, check the number of NIC if equal to qemu command.
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    if params['device_type'] == 'pci-bridge':
+        for id in range(0, 2):
```

Comment:
It's better to pass a variable to function like rang(0, num), the value of "num" could be got from cfg.